uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1303,6 +1318,22 @@ impl Attributes {
             _ => None,
         })
     }
+
+    fn validate_for_object(&self) -> Result<()> {
+        for attr in self.0.iter() {
+            match attr {
+                Attribute::Threadsafe => continue,
+                _ => bail!(format!("{:?} not supported for Objects", attr)),
+            }
+        }
+        Ok(())
+    }
+}
```

Comment:
Currently we're not validating which attributes are used for each syntactical construct. I propose we should.

Reply:
+1, yes, we should validate that e.g. you're not setting `[Throws`] on an `instance` declaration. You could consider using separate structs for this, e.g. `InstanceAttributes` and `FunctionAttributes` rather than a single `Attributes` struct, which is probably going to seem a bit verbose with this all-in-one-file approach, but would make sense if we split out the core types into their own separate files as proposed in https://github.com/mozilla/uniffi-rs/issues/286.

Reply:
Raised #373. This is worth doing, just not in this pull request. I've got a PR ready to go, but I don't want to slow down reviews.